src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java

```diff
@@ -504,20 +504,25 @@ public boolean useVariantAnnotations() {
     }
 
     /**
-     * Returns the name of the source of reads data. It can be a file name or URL.
+     * Returns the name of the source of reads data. It can be a file name or URL. Throws if the tool has more
+     * than one source.
      */
-    protected List<String> getReadSourceName(){
+    protected String getReadSourceName(){
         if (readInputs.size() > 1) {
-            throw new GATKException("Multiple ReadsDataSources specificed but a single source requested by the tool");
+            throw new GATKException("Multiple ReadsDataSources specified but a single source requested by the tool");
         }
-        return new ArrayList<>(readInputs.keySet());
+        return readInputs.keySet().stream().findFirst().get().toString();
     }
 
     /**
-     * Returns a map of read input to header.
+     * Returns the header for a given input.
      */
-    protected LinkedHashMap<String, SAMFileHeader> getReadSourceHeaderMap(){
-        return readInputs;
+    protected SAMFileHeader getHeaderForInputPath(final GATKPathSpecifier inputPathSpecifier){
```

Comment:
All I did here was to change the signature of the method so that we don't have to hand out the map. Instead the caller just queries it. So I think this change is orthogonal to the number of inputs.

Reply:
Changed the name to `getHeaderForReadsInput`.